pom.xml

```diff
@@ -290,11 +293,38 @@
         <artifactId>jacoco-maven-plugin</artifactId>
         <executions>
           <execution>
-            <id>prepare-agent</id>
+            <id>pre-unit-test</id>
             <goals>
               <goal>prepare-agent</goal>
             </goals>
           </execution>
+          <execution>
+            <id>pre-integration-test</id>
+            <goals>
+              <goal>prepare-agent-integration</goal>
+            </goals>
+            <configuration>
+              <propertyName>invoker.mavenOpts</propertyName>
+            </configuration>
+          </execution>
+          <execution>
+            <id>merge</id>
+            <goals>
+              <goal>merge</goal>
+            </goals>
+            <phase>post-integration-test</phase>
+            <configuration>
+              <fileSets>
+                <fileSet implementation="org.apache.maven.shared.model.fileset.FileSet">
+                  <directory>${project.build.directory}</directory>
+                  <includes>
+                    <include>*.exec</include>
+                  </includes>
+                </fileSet>
+              </fileSets>
+              <destFile>${project.build.directory}/jacocoAll.exec</destFile>
+            </configuration>
+          </execution>
```

Comment:
here again, some explanations ...

the `report-aggregate` is not what you believed it is (and that I believed too !). The [doc](http://www.eclemma.org/jacoco/trunk/doc/report-aggregate-mojo.html) says 

> The report is created from all modules this project depends on

So, in Jacoco you have 3 reports: the [basic](http://www.eclemma.org/jacoco/trunk/doc/report-mojo.html) one, for unit tests. The [integration](http://www.eclemma.org/jacoco/trunk/doc/report-integration-mojo.html) one, for integration tests such as ones via the invoker plugin. And the [aggregate](http://www.eclemma.org/jacoco/trunk/doc/report-aggregate-mojo.html) one.

We should ignore the aggregate one, unless you end with a Maven projet structure like this:
+ parent
|--- sortpom
|--- plugin
|--- aggregate (empty projet, depending on both sortpom and plugin, just to call the aggregate report of jacoco)

My guess is that you do not want such an empty project, so I end up with the [merge](http://www.eclemma.org/jacoco/trunk/doc/merge-mojo.html) goal of jacoco, to merge the two data files of jacoco (the one for the unit tests, the other for the integration tests) into only one file, and to use that combined file to generate the report, only one report (disabling the two others, to avoid duplication), thus having the whole coverage on the project, whatever where it comes from (unit test or integration)

Doing this also generates only one jacoco.xml, with all coverage informations from both unit and integration tests, so that the `coveralls-maven-plugin` also has the whole picture